internal/http/monitor_runner_health_test.go

```diff
@@ -0,0 +1,149 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestSendRunnerHealthCheckRequest(t *testing.T) {
+	successSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer successSrv.Close()
+
+	err := sendRunnerHealthCheckRequest(successSrv.URL)
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+
+	errorSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+	}))
+	defer errorSrv.Close()
+
+	err = sendRunnerHealthCheckRequest(errorSrv.URL)
+	if err == nil {
+		t.Error("Expected error for non-200 status code, got nil")
+	}
+}
+
+func setHealthCheckValues(t *testing.T) func() {
+	t.Helper()
+
+	originalHealthCheckInterval := healthCheckInterval
+	originalInitialDelay := initialDelay
+	originalMaxFailures := healthCheckMaxFailures
+
+	healthCheckInterval = 50 * time.Millisecond
+	initialDelay = 0
+	healthCheckMaxFailures = 2
+
+	return func() {
+		healthCheckInterval = originalHealthCheckInterval
+		initialDelay = originalInitialDelay
+		healthCheckMaxFailures = originalMaxFailures
+	}
+}
+
+func TestMonitorRunnerHealth(t *testing.T) {
+	restoreFn := setHealthCheckValues(t)
+	defer restoreFn()
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer srv.Close()
+
+	cmd := exec.Command("sleep", "1")
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		if cmd.Process != nil && cmd.Process.Kill() != nil {
+			t.Log("Failed to kill process during cleanup")
+		}
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	var wg sync.WaitGroup
+
+	MonitorRunnerHealth(ctx, cmd, srv.URL, &wg)
+
+	<-ctx.Done()
+	cancel()
```

Comment:
Is htis correct here? AFAIK `<-ctx.Done()` will block until the context is cancelled or times out. So because nobody is canceling it, until it times out. Or is the idea here that we let the timeout expire? Maybe this could be commented what all these do, and have a better name for the test caes

Reply:
Exactly, to let the timeout expire so that we verify that it respects context cancellation. My idea was to verify that the monitor starts and can monitor a process with a health endpoint, respects context cancellation, cleans up resources within timeout, and does not exceed its time limits.

Per other comment below I'll likely rewrite this to simplify.